design/object-bucket/object-bucket.md

```diff
@@ -0,0 +1,137 @@
+# ObjectBucketClaim CRD and Controller
+
+## Proposed Feature
+
+**A generalized method of S3 bucket provisioning through the implementation of an `ObjectBucketClaim` CRD and controller.** 
+
+Users will request buckets via the Kubernetes API and have returned ConfigMaps with connection information for that bucket.  
+
+One ConfigMap will be created per `ObjectBucketClaim` instance.  ConfigMap names will be deterministic and derived from the `ObjectBucketClaim` name.  This gives users the opportunity to define their Pod spec at the same time as the `ObjectBucketClaim`.  Additionally, the `ObjectBucketClaim` and Pod can be deployed at the same time thanks in-built synchronization in Kubernetes.
+
+Bucket deletion in the early stages of this controller design will not be addressed beyond cleaning up API objects.  Deletion of the bucket within the object store is left as an administrator responsibility.  This is to prevent the accidental destruction of data by users.
+
+## Design
+
+### Work Flow
```

Comment:
Could you add a section in the workflow for the admin creating the storage class that exposes the object store for bucket creation? This will help illustrate that until the admin 1) creates the storage class and 2) starts the operator that responds to the storage class, no buckets can be provisioned. This will also help explain why it is a general solution and not specific to a provider such as ceph.